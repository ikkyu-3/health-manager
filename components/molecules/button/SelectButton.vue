<template>
  <v-btn depressed fab dark :color="buttonColor" @click="click">
    <span v-if="isActive">{{ text }}</span>
  </v-btn>
</template>

<script lang="ts">
import { createComponent, computed } from '@vue/composition-api'

export default createComponent({
  props: {
    color: {
      type: String,
      default: 'red lighten-3'
    },
    isActive: {
      type: Boolean,
      default: false
    },
    text: {
      type: String,
      default: ''
    }
  },
  setup(props, ctx) {
    const buttonColor = computed(() =>
      props.isActive ? props.color : 'grey lighten-1'
    )

    function click(e: MouseEvent) {
      ctx.emit('click', e)
    }

    return { click, buttonColor }
  }
})
</script>
