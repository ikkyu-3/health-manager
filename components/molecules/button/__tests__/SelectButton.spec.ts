import Vuetify from 'vuetify'
import { mount, createLocalVue } from '@vue/test-utils'
import SelectButton from '@/components/molecules/button/SelectButton.vue'

const options = {
  localVue: createLocalVue(),
  vuetify: new Vuetify()
}

describe('molecules/button/SelectButtton.vue', () => {
  it('snapshot', () => {
    const wrapper = mount(SelectButton, {
      ...options,
      propsData: {
        color: 'red lighten-3',
        isActive: true,
        number: '1'
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  describe('color', () => {
    describe('isActiveがtrueの場合', () => {
      it('isActiveがtrueでcolorを指定しない場合、デフォルトのクラスが指定される', () => {
        const wrapper = mount(SelectButton, {
          ...options,
          propsData: {
            isActive: true
          }
        })
        expect(wrapper.find('button').classes()).toEqual(
          expect.arrayContaining(['red', 'lighten-3'])
        )
      })

      it('isActiveがtrueでcolorを指定した場合、指定したクラスが指定される', () => {
        const wrapper = mount(SelectButton, {
          ...options,
          propsData: {
            color: 'red',
            isActive: true
          }
        })
        expect(wrapper.find('button').classes()).toContain('red')
      })
    })

    describe('isActiveがfalseの場合', () => {
      it('colorを指定しても、"grey" "lighten-1"のclassが指定される', () => {
        const wrapper = mount(SelectButton, {
          ...options,
          propsData: {
            color: 'red',
            isActive: false
          }
        })
        expect(wrapper.find('button').classes()).toEqual(
          expect.arrayContaining(['grey', 'lighten-1'])
        )
      })
    })
  })

  describe('text', () => {
    it('isActiveがtureの場合、指定した文字が表示される', () => {
      const wrapper = mount(SelectButton, {
        ...options,
        propsData: { isActive: true, text: '1' }
      })
      expect(wrapper.find('.v-btn__content').text()).toBe('1')
    })

    it('isActiveがfalseの場合、指定した文字が表示されない', () => {
      const wrapper = mount(SelectButton, {
        ...options,
        propsData: { isActive: false, text: '1' }
      })
      expect(wrapper.find('.v-btn__content').text()).toBe('')
    })
  })

  it('clickした場合、@clickで登録した関数が実行される', () => {
    const click = jest.fn()
    const wrapper = mount(SelectButton, { ...options })

    wrapper.vm.$on('click', click)
    wrapper.trigger('click')

    expect(click.mock.calls).toHaveLength(1)
  })
})
