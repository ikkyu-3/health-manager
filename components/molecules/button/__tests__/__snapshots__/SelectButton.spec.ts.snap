// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`molecules/button/SelectButtton.vue snapshot 1`] = `"<button type=\\"button\\" class=\\"v-btn v-btn--depressed v-btn--fab v-btn--round theme--dark v-size--default red lighten-3\\" number=\\"1\\"><span class=\\"v-btn__content\\"><span></span></span></button>"`;
